packages/nimble-components/src/table/styles.ts

```diff
@@ -89,16 +83,77 @@ export const styles = css`
         visibility: visible;
     }
 
+    .header-container {
+        display: flex;
+        align-items: center;
+        position: relative;
+    }
+
     .header-scrollbar-spacer {
         width: var(--ni-private-table-header-scrollbar-spacer-width);
     }
 
     .header {
         flex: 1;
+        overflow: hidden;
     }
 
-    .checkbox-container {
-        display: flex;
+    .column-divider {
+        border-left: 2px solid ${popupBorderColor};
+        display: none;
+        height: ${controlSlimHeight};
+        cursor: col-resize;
+        position: absolute;
+    }
+
+    .column-divider::before {
```

Comment:
This CSS allows us to provide a larger hit-test region for the divider without adjusting its visual position. I've started with a width of 8px, which seems pretty forgiving, but am open to suggestion here.